src/apint/arithmetic.rs

```diff
@@ -407,212 +460,1060 @@ impl ApInt {
-        match self.zip_access_data_mut(rhs)? {
-            Inl(lhs, rhs) => {
-                let mut l = lhs.clone();
-                let mut r = rhs.clone();
-                l.sign_extend_from(width).unwrap();
-                r.sign_extend_from(width).unwrap();
-                let lval = l.repr() as i64;
-                let rval = r.repr() as i64;
-                let result = lval.wrapping_rem(rval) as DigitRepr;
-                *lhs = Digit(result);
-            }
-            Ext(_lhs, _rhs) => {
-                unimplemented!()
-            }
-        }
-        self.clear_unused_bits();
+        let mut rhs_clone = (*rhs).clone();
+        let negate_lhs = match ((*self).is_negative(), rhs_clone.is_negative()) {
+            (false,false) => false,
+            (true,false) => {
+                self.wrapping_neg();
+                true
+            },
+            (false,true) => {
+                rhs_clone.wrapping_neg();
+                false
+            },
+            (true,true) => {
+                self.wrapping_neg();
+                rhs_clone.wrapping_neg();
+                true
+            },
+        };
+        ApInt::wrapping_uremdiv_assign(self, &mut rhs_clone).unwrap();
+        if negate_lhs {self.wrapping_neg()}
+        //clearing unused bits is handled by `wrapping_neg()`
         Ok(())
     }
 
+    /// Divides `self` by `rhs` using **signed** interpretation and returns the remainder.
+    /// This function **may** allocate memory.
+    /// 
+    /// # Errors
+    /// 
+    /// - If `self` and `rhs` have unmatching bit widths.
+    /// - If division by zero is attempted
+    pub fn into_wrapping_srem(self, rhs: &ApInt) -> Result<ApInt> {
+        try_forward_bin_mut_impl(self, rhs, ApInt::wrapping_srem_assign)
+    }
 }
 
-// ============================================================================
-//  Standard `ops` trait implementations.
-// ----------------------------------------------------------------------------
-// 
-//  `ApInt` implements some `std::ops` traits for improved usability.
-//  Only traits for operations that do not depend on the signedness
-//  interpretation of the specific `ApInt` instance are actually implemented.
-//  Operations like `mul`, `div` and `rem` are not expected to have an
-//  implementation since a favor in unsigned or signed cannot be decided.
-// ============================================================================
-
-// ============================================================================
-//  Unary arithmetic negation: `std::ops::Add` and `std::ops::AddAssign`
-// ============================================================================
+/// ============================================================================
+///  Standard `ops` trait implementations.
+/// ----------------------------------------------------------------------------
+///  **Note:** These ops will panic if their corresponding functions return an
```

Comment:
Good note here! :)

Reply:
When I compile the with `cargo doc`, I don't see the large blocks not attached to `impl`s or functions or am I blind? Should we document this somewhere else because the trait impls are way down there in the docs?